### ISSUE TITLE
Output contains junk / zero bytes for given input and filter

### ISSUE THREAD

**Issue by nilium** [stedolan/jq#2281]
**Describe the bug**
There appears to be something causing strings to be overwritten with null bytes and garbage when jq is invoked with the given input and filter (see gist below). I unfortunately don't have much further information on where this happens in jq I can say that shortening the either the key or value in the input JSON by even one character seems to mitigates the bug.

**To Reproduce**
Input, expected output, and current output in the following gist: <https://gist.github.com/nilium/5a8bb8f89f7f81b91618243faebea9e6>

```
$ <Input.json jq -r '((tojson | "\(.)\n") | . * 5)'
```

Note the zero bytes in rows 000005f0 and 00000600 of the hexdump output, as well as what is probably junk data at rows 000007f0 and 00000800.

**Expected behavior**
The expected output is visible in the gist as Expected.json. The data itself is nonsense from reducing the input data to the smallest thing I could get to cause the bug, but there may be smaller inputs that can reproduce it.

**Environment (please complete the following information):**
 - OS and Version: Linux, Ubuntu 20.04.2 LTS
 - jq 1.6

**Additional context**
This affects the version of jq used on https://jqplay.org/ as well. It does not affect [gojq](http://github.com/itchyny/gojq).

**Comment by geirha** [comment]
On macOS with jq-1.6 installed from homebrew, I was unable to reproduce it.

On Ubuntu 20.04 with jq-1.6 installed from ubuntu repos, I was able to reproduce it.

**Comment by nilium** [comment]
Appears to segfault if run with jq-1.6 under Alpine Linux v3.13. Running in Docker:
```
: docker run -it --rm -v $(pwd):/src -w /src alpine:3.13
/src # apk add -U jq
... (2/2) Installing jq (1.6-r1) ...
/src # chmod +x Command.sh
/src # ./Command.sh
Segmentation fault (core dumped)
```
Unfortunately, the core dump doesn't appear to be useful and there are no debug symbols for jq, so that'd require more digging. (Probably later, since I have to get back to work for now.)

**Comment by geirha** [comment]
And from the same ubuntu 20.04 install, a build from tag jq-1.6 reproduced the issue, but a build from current master (jq-1.6-129-g80052e5) did not reproduce the issue, so looks like it might have gotten fixed at some point.

**Comment by nilium** [comment]
Took some time to dig around and it looks like @itchyny fixed this back in #2118. The output and behavior are similar, and it lines up with builds off of HEAD not reproducing the issue. So, I'm going to consider this already done and probably a duplicate of #2117.
